test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSourcesCommandTest.cs

```diff
@@ -41,12 +47,23 @@ public void SourcesCommandTest_AddSource(string source, bool shouldWarn)
                 CommandRunnerResult result = CommandRunner.Run(nugetexe, workingPath, string.Join(" ", args));
 
                 // Assert
-                Assert.Equal(0, result.ExitCode);
-                ISettings loadedSettings = Configuration.Settings.LoadDefaultSettings(workingPath, null, null);
-                SettingSection packageSourcesSection = loadedSettings.GetSection("packageSources");
-                SourceItem sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
-                Assert.Equal(source, sourceItem.GetValueAsPath());
-                Assert.Equal(shouldWarn, result.Output.Contains("WARNING: You are running the 'add source' operation with an 'HTTP' source"));
+                if (shouldFail)
```

Comment:
Similar feedback as https://github.com/NuGet/NuGet.Client/pull/5702#discussion_r1578665526. This feedback applies to all the occurrences of this pattern introduced in this pull-request.